sign-action.md

```diff
@@ -0,0 +1,159 @@
+# Sign an image with Notation in GitHub Actions
+
+This document walks you through how to create a GitHub Actions workflow to achieve the following goals:
+
+1. Build an image and push it to Azure Container Registry (ACR)
+2. Sign the image with Notation and Notation AKV plugin with a signing key stored in Azure Key Vault (AKV). The generated signature is automatically pushed to ACR
+
+## Prerequisites
+
+- You have created a Key Vault in AKV and created a self-signed signing key and certificate. You can follow this [doc](https://learn.microsoft.com/en-us/azure/container-registry/container-registry-tutorial-sign-build-push#create-a-self-signed-certificate-azure-cli) to create self-signed key and certificate for testing purposes. 
+- You have created a registry in Azure Container Registry
+- You have a GitHub repository to store the sample workflow file and GitHub Secrets
+
+## Create GitHub Secrets to store credentials
+
+Enter your GitHub repository, create two encrypted secrets `ACR_PASSWORD` and `AZURE_CREDENTIALS` for the repository to authenticate with ACR and AKV. See the [GitHub doc](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository) for details.
+
+### Create a GitHub Secret `ACR_PASSWORD`
+
+Create the first GitHub Secret naming `ACR_PASSWORD`, enter the password of your ACR registry where your image will be stored in the GitHub Secret value.
+
+### Create a GitHub Secret `AZURE_CREDENTIALS`
+
+Follow the steps below to get the value of `AZURE_CREDENTIALS`. This is credential to access your AKV where your signing key pair is stored.
+
+- Execute the following commands to create a new service principal on Azure. 
+
+```
+# login using your own account
+az login
+
+# Create a service principal
+spn=notationtest
+az ad sp create-for-rbac -n $spn --sdk-auth
+```
+
+> [!IMPORTANT]
+> 1. Copy the entire JSON output from the `az ad sp` command execution result into the value of GitHub Secret `AZURE_CREDENTIALS`. See [this doc](https://learn.microsoft.com/en-us/azure/developer/github/connect-from-azure?tabs=azure-portal%2Cwindows#add-the-service-principal-as-a-github-secret) for reference.
```

Comment:
At this point, the reader already knows that `AZURE_CREDENTIALS` is a GitHub Secret. We don't need to repeat it everytime.
```suggestion
> 1. Copy the entire JSON output from the `az ad sp` command execution result into the `Secret` field of `AZURE_CREDENTIALS`. See [this doc](https://learn.microsoft.com/en-us/azure/developer/github/connect-from-azure?tabs=azure-portal%2Cwindows#add-the-service-principal-as-a-github-secret) for reference.
```